### ISSUE TITLE
Replace `setExceptionPauseMode` with `setIsolatePauseMode`

### ISSUE THREAD

**Issue by annagrin** [Dart-Code/Dart-Code#4333]
**Describe the bug**
VM service intends to remove the deprecated `setExceptionPauseMode` API, so we need to replace the uses first.

```dart
  /// The `setExceptionPauseMode` RPC is used to control if an isolate pauses
  /// when an exception is thrown.
  ///
  /// mode | meaning
  /// ---- | -------
  /// None | Do not pause isolate on thrown exceptions
  /// Unhandled | Pause isolate on unhandled exceptions
  /// All  | Pause isolate on all thrown exceptions
  ///
  /// If `isolateId` refers to an isolate which has exited, then the `Collected`
  /// [Sentinel] is returned.
  ///
  /// This method will throw a [SentinelException] in the case a [Sentinel] is
  /// returned.
  @Deprecated('Use setIsolatePauseMode instead')
  Future<Success> setExceptionPauseMode(
      String isolateId, /*ExceptionPauseMode*/ String mode);
```

We are replacing the call in dwds as well.

**Comment by DanTup** [comment]
This was done in 686c268b2bf03542503881e638b19b6f4390a259 (it was raised in #4318 too), although it hasn't shipped yet. It'll be in the release towards the end of the month.

I believe the new SDK DAPs are already handling this (or rather, package:vm_service is handling it for them).

Let me know if you still think there's anything to do here!

**Comment by annagrin** [comment]
Thanks @DanTup! I encountered this while running flutter tools with the dwds master where this call is removed, it caused a hang on connecting to the debugger. I will add the method back to dwds for now until the dart-code version with the replacement is shipped. Is there an issue to monitor so we are notified when it is done?

**Comment by DanTup** [comment]
There's not a great way to be notified (since the issues are closed as they're fixed rather than released), but it will ship before the end of the month and I can ping you here once that's happened.

**Comment by DanTup** [comment]
@annagrin today I shipped new VS Code extensions (v3.58.0 for release channel, v3.59.20230115). It may take a little while for everyone to have upgraded, so when it's reasonable to land this depends on how long you want to remain compatible with older extension versions.

Generally I'd expect people to have their extensions updated more frequently than SDKs (so this is likely not a big issue, and if it does come up the fix is clear - update the VS Code extension - something that is done automatically by default). However, I have had bugs reported now and then from people that didn't update VS Code itself, which prevented the extension updated (since it has a min VS Code version), so that's something to be aware of if issues do come up after you do land it.
